src/main/kotlin/io/libp2p/mux/mplex/MplexFrameCodec.kt

```diff
@@ -54,8 +51,16 @@ class MplexFrameCodec : MessageToMessageCodec<ByteBuf, MuxFrame>() {
      */
     override fun decode(ctx: ChannelHandlerContext, msg: ByteBuf, out: MutableList<Any>) {
         while (msg.isReadable) {
+            val readerIndex = msg.readerIndex()
             val header = msg.readUvarint()
             val lenData = msg.readUvarint()
+            if (header < 0 || lenData < 0 || msg.readableBytes() < lenData) {
+                msg.readerIndex(readerIndex)
+                return
+            }
+            if (lenData > MaxMessageSize) {
```

Comment:
Why do you skip to the end of `lenData` or `messageSize`, whichever is smaller, if the `lenData` is greater than `MaxMessageSize`?

Reply:
Do you basically skip a packet if its `lenData` is greater than `MaxMessageSize`?

Reply:
Good question. I actually didn't know what the best we can do here. Just released the data so it is not accumulated further. 
We should probably at least throw exception and at most close connection since the protocol is violated without any recovery possibility

Reply:
Yeah, I’d err on the side of throwing an exception here since it’s an unsupported operation and something is clearly wrong. Closing the connection sounds a a bit harsh. That’s me assuming the connection is recoverable. i.e. we miss this one message, but do get their future messages properly. Not sure if that’s the case.

Reply:
Seems like we should log an error, and close the stream.

Reply:
I don't think we would be able to accept following data.
That most likely means the protocol is broken and we couldn't parse further data
I would just better close the connection right here, since it might be a security breach when trying to handle such situations

Reply:
What I finally did is throwing exception when `lenData > MaxMessageSize`
In the `exceptionCaught`: 
- propagate exception to let the following handlers know what's going on and why we are going to close channel. I would also prefer to keep error logging in a single place (it's now the last handler: `AbstractMuxHandler`) and not spreading it across handlers 
- `close()` the channel since if any error happens in the `decode()` method then is it likely unrecoverable. Any errors in lower level handlers (like cipher or frame length decoder) are also most likely unrecoverable